assembly/runtime.ts

```diff
@@ -677,37 +669,54 @@ export abstract class BaseContext {
   onDone_: (thiz: BaseContext) => bool;
   onDelete_: (thiz: BaseContext) => void;
 
-  constructor() {
+  constructor(context_id_: u32) {
+    this.context_id = context_id_;
     this.onDone_ = (thiz: BaseContext) => { return thiz.onDone(); };
     this.onDelete_ = (thiz: BaseContext) => { thiz.onDelete(); };
   }
 
   /**
    * Sets the effective context id to this context. this is useful for example if you receive an
    * http call in a RootContext, and want to modify headers based on the response in a regular 
-   * Context. You then will call `context.setEffectiveContext()` so that the header manipulation will
+   * Context. You then will call `setEffectiveContext(_id)` so that the header manipulation will
    * occur in the requerst context and not in the root context.
-   * the r
-   * @param effective_context_id 
+   * @param _id 
    */
-  setEffectiveContext(): WasmResultValues {
-    return imports.proxy_set_effective_context(this.context_id);
+  setEffectiveContext(_id: u32): WasmResultValues {
```

Comment:
Need a parameter because we are telling the proxy to use another context that is not the one this method is being called at.